### PR TITLE
Get file metadata from Rise Cache on 502

### PR DESCRIPTION
**Pull request by donnapep**

### CODE REVIEW

**Comment by donnapep** [comment]
@stulees @rodrigopavezi Please review. Thx.

File: rise-storage.html [hunk 1]
```diff
@@ -1051,8 +1051,13 @@
         this._hasAttemptedGetRequest = true;
       }
       else {
-        this._startTimer();
-        this.fire("rise-cache-error", resp);
+        // Get metadata for single file on a 502 response from Rise Cache.
+        if ((e.detail.request.xhr.response.status === 502) && this._isFile) {
+          this.getMetadata();
```
Comment (stulees) [inline rise-storage.html]: @donnapep Should this be getting metadata via RC and not storage directly?

Comment (donnapep) [inline rise-storage.html]: @stulees It does. This call will eventually end up here - https://github.com/Rise-Vision/rise-storage/blob/763e8a757a81d9fee7ac789f3c6bfbe4be2952a0/rise-storage.html#L499

Comment (stulees) [inline rise-storage.html]: @donnapep Ah right, thanks for reminding me.

**Comment by stulees** [comment]
@donnapep 👍

File: rise-storage.html [hunk 2]
```diff
@@ -1051,8 +1051,13 @@
         this._hasAttemptedGetRequest = true;
       }
       else {
-        this._startTimer();
-        this.fire("rise-cache-error", resp);
+        // Get metadata for single file on a 502 response from Rise Cache.
+        if ((e.detail.request.xhr.response.status === 502) && this._isFile) {
```
Comment (rodrigopavezi) [inline rise-storage.html]: @donnapep I think it would be good to call metadata for 5xx status code and not only for 502. What do you think? cheers

**Pull request merged**